dev/com.ibm.ws.repository.resolver/src/com/ibm/ws/repository/resolver/RepositoryResolver.java

```diff
@@ -900,11 +1005,25 @@ private EsaResource getResource(ProvisioningFeatureDefinition feature) {
      * @throws RepositoryResolutionException if any errors occurred during resolution
      */
     private void reportErrors() throws RepositoryResolutionException {
-        if (resourcesWrongProduct.isEmpty() && missingTopLevelRequirements.isEmpty() && missingRequirements.isEmpty() && featureConflicts.isEmpty()) {
+        if (resourcesWrongProduct.isEmpty() && missingTopLevelRequirements.isEmpty() && missingRequirements.isEmpty() && featureConflicts.isEmpty() && !hasVersionlessIssue()) {
             // Everything went fine!
             return;
         }
 
+        List<String> missingBasePlatforms = new ArrayList<String>();
+
+        // Versionless feature issues will appear in missingTopLevelRequirements, and this will gather the associated platform unable to target.
+        for (String name : missingTopLevelRequirements) {
+            ProvisioningFeatureDefinition feature = resolverRepository.getFeature(name);
+            if (feature != null && feature.isVersionless()) {
+                ProvisioningFeatureDefinition firstChild = resolverRepository.findAllPossibleVersions(feature).get(0);
+                String plat = firstChild.getPlatformName();
```

Comment:
do we need to null check firstChild?

Reply:
If we do, we need to check whether the list is empty before calling `get(0)` because that will throw an exception.

Reply:
Agreed made changes to check